inst/include/Rcpp/Nullable.h

```diff
@@ -116,6 +116,16 @@ namespace Rcpp {
          */
         inline bool isSet(void) const { return m_set; }
 
+        /**
+         * Returns m_sexp as a T
+         */
+        inline T as() { return Rcpp::as< T >(get()); }
+
+        /**
+         * Return a clone of m_sexp as a T
+         */
+        inline T clone() const { return Rcpp::clone(Rcpp::as< T >(get())); }
```

Comment:
So technically the call is just AWFUL it:

* Shallow copies in the return from `Rcpp::as()`
* Deep copies inside the `Rcpp::clone()' function
* Shallow copies in the return from `Rcpp::clone()`
* Shallow copies in the return from `Nullable::clone()`
* Probably shallow copies again when the end user receives it.

There should really be a better way.  Best I can come up with is a `SEXP Rcpp::clone(SEXP obj)` function which would remove the need for some of this.

Any other thoughts?